domain/ticket-domain/src/main/java/com/uket/domain/ticket/repository/TicketRepository.java

```diff
@@ -28,5 +32,14 @@ public interface TicketRepository extends JpaRepository<Ticket,Long> {
     List<Ticket> findAllByUserIdAndStatusNot(Long userId, TicketStatus status);
 
     void deleteAllByUserId(Long userId);
+
+    @Query("SELECT t FROM Ticket t " +
+        "WHERE t.user.id = :userId " +
+        "AND t.status NOT IN (:cancelled, :expired) " +
+        "AND (t.event.endDate > :now)")
+    List<Ticket> findValidTicketsByUserId(@Param("userId") Long userId,
```

Comment:
```suggestion
    List<Ticket> findAllByUserIdAndStatusNotInAndEventEndDateAfter(Long userId, List<TicketStatus> status, LocalDate now);
```
요렇게도 되긴합니다! 쿼리를 직접 쓰는 것도 메서드명을 짧게 가져갈 수 있어서 좋은 것 같긴해용 👍 👍 
ticketStatus만 List로 받도록 수정하면 좋을 것 같습니다!

Reply:
메서드명은 짧게 가져가고싶어서 query 수정 및 List로 ticketStatus받을 수 있도록 수정 구현했습니다!